hw/ip/lc_ctrl/lc_ctrl.core

```diff
@@ -8,6 +8,7 @@ description: "LC Controller"
 filesets:
   files_rtl:
     depend:
+      - lowrisc:constants:jtag_id_pkg
```

Comment:
LC_CTRL does not appear to use the parameter either. It's probably better for the IPs to not reference this, and let the value be set by the top level instead.

Reply:
right i added these since the expectation is that rv_dm and lc_ctrl will make use of them eventually.

Reply:
@tjaychen 
Just reminding about the foundry CL. Please approve it once this PR is merged or our nightly regression tests will fail...

Thanks

Reply:
So will the top level no longer override the id code value like it is done here? 
https://github.com/lowRISC/opentitan/blob/2023653b220d6fd4dde5fd9055a6ee5eb0802f2b/hw/top_earlgrey/rtl/autogen/top_earlgrey.sv#L2106

Hmm, interestingly `top_earlgrey` currently overrides the value on RV_DM (that too, with `RvDmIdCodeValue`, not `JTAG_IDCODE`), but not the LC_CTRL, which seems like a bug.

Reply:
i feel like maybe we should change to directly referencing this value inside rv_dm. 
@msfschaffner

Reply:
@sriyerg yeah we need to fix that. this is one of the reasons why #11616 got filed.

@tjaychen I am not sure... isn't this a top-level-specific property that we want to propagate down to the block level when instantiating the IPs (because every top-level / chip instance might have a slightly different JTAG config)? I know we already have dependencies from top-level-specific cores down into some of our IPs, but I think it would be cleaner to avoid that in general...

Reply:
Aah, sorry I didn't realize it was already pointed out. I tend to also think it is better for the top level to drive this value, so the IPs can actually remain oblivious to it (and as a plus, minimize the instances of fusesoc flag based file swapping).

Reply:
but wouldn't the partners be required to swap the id package file? 
So for a specific chip instance it would indeed be different. 

and yes i think for another top_* directory would have a different value. 
But this ties back to a long standing issue of how we intend to support different tops.

Reply:
my most preferred approach is honestly to feed this directly from a propreitary module and avoid having to switch parameter declarations.  That's why I originally suggested this be driven from the chip_level_* via AST.

Reply:
> but wouldn't the partners be required to swap the id package file? So for a specific chip instance it would indeed be different.
> 
> and yes i think for another top_* directory would have a different value. But this ties back to a long standing issue of how we intend to support different tops.

sorry I may have described this inaccurately. I wasn't suggesting to remove the fusesoc switch, since as you point out partners will need to be able to swap the id package file (`lowrisc:constants:jtag_id_pkg`). what I was suggesting was to only make the top-level design dependent on that id package, and then feed the idcodes defined in that package into `rv_dm` and `lc_ctrl` via an instantiation parameter instead of having `rv_dm` and `lc_ctrl` depend on...

Reply:
ah i see. that makes sense. so you mean in the parameter declaration directly reference this?
That is cleaner, let me do that.

Reply:
Right, when I said "minimize the instances of fusesoc flag based file swapping", I meant limit it only to `top_earlgrey.core`.